src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -73,6 +74,9 @@ public class FileChannelImpl
     // Used to make native read and write calls
     private static final FileDispatcher nd = new FileDispatcherImpl();
 
+    // Flag that determines if file reads/writes should be traced by JFR
+    private static boolean jfrTracing;
```

Comment:
Should the force method be changed to test this flag too?

I'm also wondering about the transferXXX methods. We might want to think about these for a separate PR as they have more potential to be outliers than the read/write methods.

Reply:
I think it would be good to use the flag for all events, but I rather do it as separate PR so this change is mostly mechanical change to remove ASM. It makes it easier to track regressions or improvements.

I can file an enhancement for the transferXXX methods.

Reply:
"JFR: Add file event support for transfer methods"
https://bugs.openjdk.org/browse/JDK-8331995

"JFR: Add boolean check before loading event classes"
https://bugs.openjdk.org/browse/JDK-8331996